utils/dgcClientSelector.ts

```diff
@@ -0,0 +1,40 @@
+import { Client as DGCClient } from "../generated/dgc/client";
+import { IConfig } from "./config";
+import { toHash } from "./hash";
+
+/**
+ * Defines a selector object of a type T.
+ * A selector returns different implementations of T depending on the value of the fiscal code provided
+ */
+interface ISelector<T> {
+  readonly select: (hashedFiscalCode: string) => T;
+}
+
+/**
+ * Build a selector object for DGC http client
+ *
+ * @param param0
+ * @returns
+ */
+export const createDGCClientSelector = ({
```

Comment:
I think we could be in conflict with this implementations:
https://github.com/pagopa/io-functions-eucovidcerts/pull/8/files/7721689a23bb74489b6f298eeb7c76496a59c7a0#diff-6f604771796a8d31c3588d4f1be37a76dbe0fa0ee2bb88db00fdeeb48ff9eff7
We have to merge them in a unique function.